modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -647,6 +652,56 @@ def base_map_partitions(
             ]
         )
 
+    @classmethod
+    @wait_computations_if_benchmark_mode
+    def broadcast_apply(
+        cls,
+        axis,
+        apply_func,
+        left,
+        right,
+    ):
+        """
+        Broadcast the `right` partitions to `left` and apply `apply_func` function using different approaches to achieve the best performance.
+
+        Parameters
+        ----------
+        axis : {0, 1}
+            Axis to apply and broadcast over.
+        apply_func : callable
+            Function to apply.
+        left : np.ndarray
+            NumPy array of left partitions.
+        right : np.ndarray
+            NumPy array of right partitions.
+
+        Returns
+        -------
+        np.ndarray
+            NumPy array of result partition objects.
+        """
+        # The `broadcast_apply` runtime condition differs from
+        # the same condition in `map_partitions` because the columnar
+        # approach for `broadcast_apply` results in a slowdown.
+        if np.prod(left.shape) <= 1.5 * CpuCount.get():
```

Comment:
Now we have the same condition, haven't we?

Reply:
Not really, because in `map_partitions` we use 3 approaches: `base_map_partitions`, `map_partitions_joined_by_column` and `map_axis_partitions`.
In `broadcast_apply` we use only 2 approaches: `base_broadcast_apply` and `broadcast_axis_partitions` because `broadcast_joined_by_axis` does not work as well as for map.

Reply:
This condition changed using a new environment variable `DynamicPartitioning`